### PR TITLE
Implement tests for PR #512

### PR DESCRIPTION
**Pull request by wmitzel-airplus**
Tests for AADAppRoleAuthenticationFilter and UserPrincipalManagerAudience.

## Summary
This PR contains the extracted unit and system tests of Pull Request #512, as requested during the review.

## Issue Type
- Tests for a new feature

## Starter Names
  - active directory spring boot starter

## Additional Information
n/a

### CODE REVIEW

**Comment by wmitzel-airplus** [comment]
I've updated the tests to the new filter name. This PR is ready for review now.

File: azure-spring-boot/src/test/java/com/microsoft/azure/spring/autoconfigure/aad/AADAppRoleAuthenticationFilterTest.java [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+            }
+        };
+
+        filter.doFilterInternal(request, response, filterChain);
+
+        verify(userPrincipalManager).buildUserPrincipal(TOKEN);
+        assertNull("Authentication has not been cleaned up!", SecurityContextHolder.getContext().getAuthentication());
+    }
+
+
```
Comment (Incarnation-p-lee) [inline azure-spring-boot/src/test/java/com/microsoft/azure/spring/autoconfigure/aad/AADAppRoleAuthenticationFilterTest.java]: duplicated empty line

File: azure-spring-boot/src/test/java/com/microsoft/azure/spring/autoconfigure/aad/UserPrincipalManagerAudienceTest.java [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+import java.util.Date;
+import org.junit.Before;
+import org.junit.Test;
+
+public class UserPrincipalManagerAudienceTest {
+
+  private static final String FAKE_CLIENT_ID = "dsflkjsdflkjsdf";
+  private static final String FAKE_APPLICATION_URI = "https://oihiugjuzfvbhg";
+
+
```
Comment (Incarnation-p-lee) [inline azure-spring-boot/src/test/java/com/microsoft/azure/spring/autoconfigure/aad/UserPrincipalManagerAudienceTest.java]: duplicated empty line.

File: azure-spring-boot/src/test/java/com/microsoft/azure/spring/autoconfigure/aad/UserPrincipalManagerAudienceTest.java [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+    resourceRetriever = url -> new Resource(jwkString, "application/json");
+
+    serviceEndpointsProperties = mock(ServiceEndpointsProperties.class);
+    aadAuthenticationProperties = new AADAuthenticationProperties();
+    aadAuthenticationProperties.setClientId(FAKE_CLIENT_ID);
+    aadAuthenticationProperties.setAppIdUri(FAKE_APPLICATION_URI);
+    final ServiceEndpoints serviceEndpoints = new ServiceEndpoints();
+    serviceEndpoints.setAadKeyDiscoveryUri("file://dummy");
+    when(serviceEndpointsProperties.getServiceEndpoints(anyString())).thenReturn(serviceEndpoints);
+
```
Comment (Incarnation-p-lee) [inline azure-spring-boot/src/test/java/com/microsoft/azure/spring/autoconfigure/aad/UserPrincipalManagerAudienceTest.java]: unnecessary empty line ?

File: azure-spring-boot/src/test/java/com/microsoft/azure/spring/autoconfigure/aad/UserPrincipalManagerAudienceTest.java (hunk 2 above)
Comment (wmitzel-airplus) [reply azure-spring-boot/src/test/java/com/microsoft/azure/spring/autoconfigure/aad/UserPrincipalManagerAudienceTest.java]: Fixed

File: azure-spring-boot/src/test/java/com/microsoft/azure/spring/autoconfigure/aad/UserPrincipalManagerAudienceTest.java (hunk 3 above)
Comment (wmitzel-airplus) [reply azure-spring-boot/src/test/java/com/microsoft/azure/spring/autoconfigure/aad/UserPrincipalManagerAudienceTest.java]: Fixed

**Comment by wmitzel-airplus** [comment]
Thanks for the review. I've removed the superfluous blank lines.

**Comment by Incarnation-p-lee** [comment]
Thanks a lot for your great work.

**Pull request merged**